packages/babel-plugin-transform-react-jsx-source/src/index.js

```diff
@@ -10,32 +10,43 @@
  *
  * becomes:
  *
- * <sometag __source={{fileName: 'this/file.js', lineNumber: 10}}/>
+ * var __jsxFileName = 'this/file.js';
+ * <sometag __source={{fileName: __jsxFileName, lineNumber: 10}}/>
  */
 
 
-import path from "path";
-
 const TRACE_ID = "__source";
+const FILE_NAME_VAR = "__jsxFileName";
 
 export default function ({ types: t }) {
-  function makeTrace(fileName, lineNumber) {
-    const fileNameLiteral = fileName != null ? t.stringLiteral(fileName) : t.nullLiteral();
+  function makeTrace(lineNumber) {
     const fileLineLiteral = lineNumber != null ? t.numericLiteral(lineNumber) : t.nullLiteral();
-    const fileNameProperty = t.objectProperty(t.identifier("fileName"), fileNameLiteral);
+    const fileNameProperty = t.objectProperty(t.identifier("fileName"), t.identifier(FILE_NAME_VAR));
```

Comment:
Might be better to generate a unique identifier for this file. A bit safer. E.g. if a file gets transformed twice for whatever reason or one transformed file is copy/pasted into a static file.

https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#generating-a-uid